### PR TITLE
[GLUTEN-3221][CH]fix readDecimal func for r2c when reading negative decimal data

### PR DESCRIPTION
**Pull request by lhuang09287750**
## What changes were proposed in this pull request?

pad buffer with '\xff' other than 0 when reading negative decimal data during r2c

(Fixes: \#3221)

## How was this patch tested?

add a new test case in GlutenDataFrameAggregateSuite

### CODE REVIEW

**Comment by lhuang09287750** [comment]
retest this please

File: cpp-ch/local-engine/Parser/SparkRowToCHColumn.cpp [hunk 1]
```diff
@@ -164,6 +164,18 @@ Field VariableLengthDataReader::readDecimal(const char * buffer, size_t length)
     assert(sizeof(Decimal128) >= length);
 
     char decimal128_fix_data[sizeof(Decimal128)] = {};
+
+    char * tmp_buf = new char[length];
```
Comment (taiyang-li) [inline cpp-ch/local-engine/Parser/SparkRowToCHColumn.cpp:168]: you forgot to freee it .

**Review by taiyang-li — CHANGES_REQUESTED** [review]

File: cpp-ch/local-engine/Parser/SparkRowToCHColumn.cpp (hunk 1 above)
Comment (taiyang-li) [reply cpp-ch/local-engine/Parser/SparkRowToCHColumn.cpp:168]: 这里又增加了一个buffer的创建、复制、销毁，外加一个String的构造、swap和销毁，开销不小，可能会影响性能。是否可以在不创建额外buffer的情况下，确定对应的decimal value是否为negative ?

**Review by taiyang-li — CHANGES_REQUESTED** [review]

**Comment by lhuang09287750** [comment]
@taiyang-li 收到，我试试不用buffer来判断负的decimal

**Comment by lhuang09287750** [comment]
@taiyang-li 改了判断方法，因为decimal value存的是对应的整型的补码，直接用其包含最高位（符号位）的那个字节转成int8来判断 整体decimal的正负。

**Review by taiyang-li — APPROVED** [review]
LGTM

**Review by taiyang-li — APPROVED** [review]

**Comment by lhuang09287750** [comment]
fixed in my anthor pr,close!

**Pull request closed without merging**